transport-native-unix-common/src/main/java/io/netty/channel/unix/Socket.java

```diff
@@ -257,6 +268,18 @@ public final boolean finishConnect() throws IOException {
         return true;
     }
 
+    public final boolean disconnect() throws IOException {
+        int res = disconnect(fd);
+        if (res < 0) {
+            if (res == ERRNO_EINPROGRESS_NEGATIVE) {
```

Comment:
when would you expect this to happen, and wouldn't the disconnect cancel any pending connect?